xmake/toolchains/rust/xmake.lua

```diff
@@ -28,58 +28,16 @@ toolchain("rust")
     set_toolset("rcar", "$(env RC)", "rustc")
 
     on_load(function (toolchain)
-        local target
-        if toolchain:is_arch("x86_64", "x64") then
-            target = "x86_64"
-        elseif toolchain:is_arch("i386", "x86", "i686") then
-            target = "i686"
-        elseif toolchain:is_arch("arm64", "aarch64", "arm64-v8a") then
-            target = "aarch64"
-        elseif toolchain:is_arch("armeabi-v7a", "armv7-a") then
-            target = "armv7"
-        elseif toolchain:is_arch("armeabi", "armv5te") then
-            target = "arm"
-        elseif toolchain:is_arch("wasm32") then
-            target = "wasm32"
-        elseif toolchain:is_arch("wasm64") then
-            target = "wasm64"
-        end
+        import("private.tools.rust.target_triple")
 
-        if target then
-            if toolchain:is_plat("windows") then
-                target = target .. "-pc-windows-msvc"
-            elseif toolchain:is_plat("mingw") then
-                target = target .. "-pc-windows-gnu"
-            elseif toolchain:is_plat("linux") then
-                target = target .. "-unknown-linux-gnu"
-            elseif toolchain:is_plat("macosx") then
-                target = target .. "-apple-darwin"
-            elseif toolchain:is_plat("android") then
-                target = target .. "-linux-"
-                if toolchain:is_arch("armeabi-v7a", "armeabi", "armv7-a", "armv5te") then
-                    target = target .. "androideabi"
-                else
-                    target = target .. "android"
-                end
-            elseif toolchain:is_plat("iphoneos", "appletvos", "watchos") then
-                if toolchain:is_plat("iphoneos") then
-                    target = target .. "-apple-ios"
-                elseif toolchain:is_plat("appletvos") then
-                    target = target .. "-apple-tvos"
-                elseif toolchain:is_plat("watchos") then
-                    target = target .. "-apple-watchos"
-                end
-                if toolchain:config("appledev") == "simulator" then
-                    target = target .. "-sim"
-                end
-            elseif toolchain:is_plat("bsd") then
-                target = target .. "-unknown-freebsd"
-            elseif toolchain:is_plat("wasm") then
-                target = target .. "-unknown-unknown"
-            end 
+        local opt = {}
+        if toolchain:config("appledev") == "simulator" then
+            opt.apple_sim = true
         end
-
+ 
+        local target = target_triple(toolchain:plat(), toolchain:arch(), opt)
         if target then
+            toolchain:add("rcflags", "--verbose")
```

Comment:
why? if we want to get verbose info, we should use `option.get("verbose")` and `option.get("diagnosis")`

Reply:
It's a leftover, I'm removing it